File: PolicyPlus/EditPolValue.vb
Imports Microsoft.Win32
Public Class EditPolValue
    Public SelectedKind As RegistryValueKind
    Public ChosenName As String
    Private Sub EditPolValueType_KeyDown(sender As Object, e As KeyEventArgs) Handles Me.KeyDown
        If e.KeyCode = Keys.Escape Then DialogResult = DialogResult.Cancel
    End Sub
    Public Function PresentDialog() As DialogResult
        TextName.Text = ""
        ComboKind.SelectedIndex = 0
        Dim dlgRes = ShowDialog()
        If dlgRes = DialogResult.OK Then
            Select Case ComboKind.SelectedIndex
                Case 0
                    SelectedKind = RegistryValueKind.String
                Case 1
                    SelectedKind = RegistryValueKind.ExpandString
                Case 2
                    SelectedKind = RegistryValueKind.MultiString
                Case 3
                    SelectedKind = RegistryValueKind.DWord
                Case 4
                    SelectedKind = RegistryValueKind.QWord
            End Select
            ChosenName = TextName.Text
        End If
        Return dlgRes
    End Function
End Class

File: PolicyPlus/EditPolValue.cs
using System.Windows.Forms;
using Microsoft.Win32;

namespace PolicyPlus
{
    public partial class EditPolValue
    {
        public RegistryValueKind SelectedKind;
        public string ChosenName;

        public EditPolValue()
        {
            InitializeComponent();
        }
        private void EditPolValueType_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
                DialogResult = DialogResult.Cancel;
        }
        public DialogResult PresentDialog()
        {
            TextName.Text = "";
            ComboKind.SelectedIndex = 0;
            var dlgRes = ShowDialog();
            if (dlgRes == DialogResult.OK)
            {
                switch (ComboKind.SelectedIndex)
                {
                    case 0:
                        {
                            SelectedKind = RegistryValueKind.String;
                            break;
                        }
                    case 1:
                        {
                            SelectedKind = RegistryValueKind.ExpandString;
                            break;
                        }
                    case 2:
                        {
                            SelectedKind = RegistryValueKind.MultiString;
                            break;
                        }
                    case 3:
                        {
                            SelectedKind = RegistryValueKind.DWord;
                            break;
                        }
                    case 4:
                        {
                            SelectedKind = RegistryValueKind.QWord;
                            break;
                        }
                }
                ChosenName = TextName.Text;
            }
            return dlgRes;
        }
    }
}